src/pages/workspaces/workspace/analysis/modals/ComputeModal.js

```diff
@@ -1756,14 +1756,18 @@ export const ComputeModalBase = ({
     ])
   }
 
-  const renderPersistentDiskSection = diskExists => {
+  const diskSizeId = useUniqueId()
+  const persistentDiskId = useUniqueId()
```

Comment:
It would be really good to minimize the scope of these variables. For instance, can you move line 1760 (`persistentDiskId`) down to line 1763 so that it is within `RenderPersistentDiskSection`?

Reply:
I wish that were the case too, but it causes render looping errors on the tests and on the application when running. It'll have to be something we fix when we eventually convert this to a ts file

Reply:
The reason is that you haven't changed these sections to React components, and instead they are still just "render methods". A component looks like this (I don't think the dict input argument is technically necessary, but ours are often of this format):

```
const PersistentDiskType = ({ diskExists }) => {
    const persistentDiskId = useUniqueId()
    return h(div, [
```

and you use it like this:

```
side: 'bottom'}, [h(PersistentDiskType, { diskExists} )])
```

If you are instead doing something like the code snippet below, you are not rendering a component-- you are just calling a helper method that returns some HTML.

```
side: 'bottom'}, [RenderPersistentDiskType()]) 
```

Reply:
yes, what @cahrens said.  As soon as I saw the hook calls OUTSIDE in the module scope, I knew something was wrong... This will all be so much easier to avoid confusion once we have broader Typescript adoption. But yes, follow Christina's guidance.

Reply:
also, IA team should definitely get a ticket on the backlog to break this file up and convert to TS... Nearly 2000 lines of code is VERY big.... I see lots and lots of renderXxxx helper methods that should be broken out into proper functional components... I would push ComputeModal file into a subfolder of the same name, then start yanking out these smaller components into adjascent files.... and add some Typescript at the same time :)

This file is near or at the top of SonarCube's recent complexity index scan BTW.

Reply:
I often find myself drowning in the complexity of the file, I'd appreciate it very much if someone fixed it. I'll try to have this change out soon, just trying to resolve some failing tests after applying @cahrens helpful suggestion